jerry-core/jerry.c

```diff
@@ -591,6 +591,38 @@ jerry_get_array_index_value (jerry_object_t *array_obj_p, /**< array object */
 } /* jerry_get_array_index_value */
 
 /**
+ * Get length of an array object
+ *
+ * @return length of the given array
+ */
+jerry_length_t
+jerry_get_array_length (const jerry_object_t *object_p)
+{
+  jerry_assert_api_available ();
+
+  JERRY_ASSERT (jerry_is_array (object_p));
+
+  jerry_length_t length = 0;
+  ecma_string_t *magic_string_length_p = ecma_get_magic_string (LIT_MAGIC_STRING_LENGTH);
+  ecma_value_t ret_value = ecma_make_simple_value (ECMA_SIMPLE_VALUE_EMPTY);
+
+  ECMA_TRY_CATCH (len_value,
+                  ecma_op_object_get ((jerry_object_t *) object_p, magic_string_length_p),
+                  ret_value);
+  ECMA_OP_TO_NUMBER_TRY_CATCH (len_number, len_value, ret_value);
+
+  length = ecma_number_to_uint32 (len_number);
+
+  ECMA_OP_TO_NUMBER_FINALIZE (len_number);
+  ECMA_FINALIZE (len_value);
+
+  ecma_deref_ecma_string (magic_string_length_p);
+  JERRY_ASSERT (!ECMA_IS_VALUE_ERROR (ret_value));
```

Comment:
It would be better to handle/free the error if we happen to get one. I don't think hoping that people will play by the rules, and only pass valid arrays to this function is enough.